src/Parser/Hunter/Marksmanship/Modules/Talents/PatientSniper/PatientSniperDetails.js

```diff
@@ -0,0 +1,97 @@
+    abilityTracker: AbilityTracker,
+  };
+
+  hasPiercingShot = false;
+  vulnerableModifier = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.PATIENT_SNIPER_TALENT.id);
+    this.hasPiercingShot = !!this.combatants.selected.hasTalent(SPELLS.PIERCING_SHOT_TALENT.id);
+    const unerringArrowsRank = this.combatants.selected.traitsBySpellId[SPELLS.UNERRING_ARROWS_TRAIT.id];
+    this.vulnerableModifier = VULNERABLE_BONUS + unerringArrowsRank * UNERRING_ARROWS_BONUS_PER_RANK;
+  }
+
+  get bonusAimedDamage() {
+    return this.patientSniperTracker.patientSniper[SPELLS.AIMED_SHOT.id].bonusDmg;
+  }
+  get bonusPiercingDamage() {
+    return this.patientSniperTracker.patientSniper[SPELLS.PIERCING_SHOT_TALENT.id].bonusDmg;
+  }
+  get bonusDamage() {
+    return this.bonusAimedDamage + this.bonusPiercingDamage;
+  }
+
+  suggestions(when) {
+    const bonusDamagePercentage = this.owner.getPercentageOfTotalDamageDone(this.bonusDamage);
+    const MINOR = 0.18;
+    const AVG = 0.17;
+    const MAJOR = 0.15;
+    when(bonusDamagePercentage).isLessThan(MINOR)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span><SpellLink id={SPELLS.PATIENT_SNIPER_TALENT.id}/> increases the damage of your <SpellLink id={SPELLS.AIMED_SHOT.id}/> or <SpellLink id={SPELLS.PIERCING_SHOT_TALENT.id}/> the later you fire them inside <SpellLink id={SPELLS.VULNERABLE.id}/>. While this isn't worth waiting for, it looks like you're shooting your Aimed Shots / Piercing Shots too soon, try and use <SpellLink id={SPELLS.ARCANE_SHOT.id}/> as a filler after applying Vulnerable. If you have enough haste you can fit in two Arcane Shots instead of one.</span>)
+          .icon('ability_hunter_snipertraining')
+          .actual(`${formatPercentage(actual)}% bonus damage`)
+          .recommended(`> ${formatPercentage(recommended)}% bonus damage is recommended`)
+          .regular(AVG)
+          .major(MAJOR);
+      });
+  }
+
+  statistic() {
+    let tooltipText = `Your Aimed Shots ${this.hasPiercingShot ? 'and Piercing Shots ' : ''}did ${formatNumber(this.bonusDamage)} (${this.owner.formatItemDamageDone(this.bonusDamage)}) bonus damage thanks to Patient Sniper talent. Below you'll see them individually, and if you want to see more Patient Sniper information (such as without Trueshot windows), please check the "Patient Sniper Usage" tab in the menu. <br />`;
+    const aimed = this.abilityTracker.getAbility(SPELLS.AIMED_SHOT.id);
+    tooltipText += `Aimed Shot bonus damage: ${formatNumber(this.bonusAimedDamage)} (${formatPercentage(this.bonusAimedDamage / aimed.damageEffective)} %)`;
+    if (this.hasPiercingShot) {
+      const piercing = this.abilityTracker.getAbility(SPELLS.PIERCING_SHOT_TALENT.id);
+      tooltipText += `<br />Piercing Shot bonus damage: ${formatNumber(this.bonusPiercingDamage)} (${formatPercentage(this.bonusPiercingDamage / piercing.damageEffective)} %)`;
+    }
+    return (
```

Comment:
Shouldn't this tooltipText be for both Piercing AND Aimed ? or are they still together if you have piercing shot selected?

Reply:
That's why there's tooltipText += ..., it adds the Piercing Shot part if you have it talented but the aimed part stays there regardless. I'll add screenshots later.

Reply:
With Piercing Shot:
![image](https://user-images.githubusercontent.com/5068584/31854656-17b27a38-b69d-11e7-941a-ba7f7166c1ee.png)


Without Piercing Shot:
![image](https://user-images.githubusercontent.com/5068584/31854641-e8a5ca2e-b69c-11e7-9958-17b88734d880.png)